internal/engine/buildcontroller_test.go

```diff
@@ -1266,6 +1266,92 @@ func TestErrorHandlingWithMultipleBuilds(t *testing.T) {
 	f.assertAllBuildsConsumed()
 }
 
+func TestManifestsWithSameTwoImages(t *testing.T) {
+	f := newTestFixture(t)
+	m1, m2 := NewManifestsWithSameTwoImages(f)
+	f.Start([]model.Manifest{m1, m2})
+
+	f.waitForCompletedBuildCount(2)
+
+	call := f.nextCall("m1 build1")
+	assert.Equal(t, m1.K8sTarget(), call.k8s())
+
+	call = f.nextCall("m2 build1")
+	assert.Equal(t, m2.K8sTarget(), call.k8s())
+
+	aPath := f.JoinPath("common", "a.txt")
+	f.fsWatcher.Events <- watch.NewFileEvent(aPath)
+
+	f.waitForCompletedBuildCount(4)
+
+	// Make sure that both builds are triggered, and that they
+	// are triggered in a particular order.
+	call = f.nextCall("m1 build2")
+	assert.Equal(t, m1.K8sTarget(), call.k8s())
+
+	imageStateM1I0 := call.state[m1.ImageTargets[0].ID()]
+	assert.Equal(t, map[string]bool{aPath: true}, imageStateM1I0.FilesChangedSet)
+
+	call = f.nextCall("m2 build2")
+	assert.Equal(t, m2.K8sTarget(), call.k8s())
+
+	// Make sure that when the second build is trigged, we did the bookkeeping
+	// correctly around marking the first and second image built, and only rebuilding
+	// the third image.
+	idM2I0 := m2.ImageTargets[0].ID()
+	idM2I1 := m2.ImageTargets[1].ID()
+	assert.Equal(t, map[string]bool{}, call.state[idM2I0].FilesChangedSet)
```

Comment:
a. i'm surprised not to see an assertion about `state[idM2I0].DepsChanged` or `state[idM2I1].FilesChanged`, but also
b. maybe worth verifying that `call.state` contains `idM2I0` and `idM2I`--if for some reason it doesn't, these checks will still pass.
c. fwiw these names are hard to read, i have to squint at this code to be able to tell what's going on